fine_tuning_mistral_beavertails/beavertails.ipynb

```diff
@@ -0,0 +1,5831 @@
+            "5d5c85a5ff6d4b9ea78bc7d441f6a6cc",
+            "302bca731896426e95882adf15035012",
+            "e37af2db28124bb3ac7d63eaa4769df0",
+            "ec6e96f0156d485a88f231b24692a917",
+            "0b300776f06c4cd9a2aaf3616a1f54dd",
+            "c6612e5393074ddf88b1a93a27b0b7b3"
+          ]
+        },
+        "id": "Fxn6llbHqN0s",
+        "outputId": "1c968369-9fac-4f38-b614-3d0efc8de29c"
+      },
+      "outputs": [
+        {
+          "data": {
+            "application/vnd.jupyter.widget-view+json": {
+              "model_id": "b679396c8bed4e01b09f9151038146f4",
+              "version_major": 2,
+              "version_minor": 0
+            },
+            "text/plain": [
+              "Loading checkpoint shards:   0%|          | 0/2 [00:00<?, ?it/s]"
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        },
+        {
+          "name": "stderr",
+          "output_type": "stream",
+          "text": [
+            "Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.\n"
+          ]
+        }
+      ],
+      "source": [
+        "# BitsAndBytesConfig int-4 config\n",
+        "bnb_4bit_compute_dtype=\"float16\"\n",
+        "compute_dtype = getattr(torch, bnb_4bit_compute_dtype)\n",
+        "bnb_config = BitsAndBytesConfig(\n",
+        "    load_in_4bit=True,\n",
+        "    bnb_4bit_use_double_quant=False,\n",
+        "    bnb_4bit_quant_type=\"nf4\",\n",
+        "    bnb_4bit_compute_dtype=compute_dtype,\n",
+        ")\n",
+        "\n",
+        "# Load model and tokenizer\n",
+        "model = AutoModelForCausalLM.from_pretrained(\n",
+        "    mistral_checkpoint,\n",
+        "    quantization_config=bnb_config,\n",
+        "    use_cache=False,\n",
+        "    device_map={\"\":0},\n",
+        ")\n",
+        "model.config.pretraining_tp = 1\n",
+        "\n",
+        "tokenizer = AutoTokenizer.from_pretrained(mistral_checkpoint)\n",
+        "tokenizer.pad_token = tokenizer.eos_token\n",
+        "tokenizer.padding_side = \"right\""
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "cPsLcs-xCBfG"
+      },
+      "source": [
+        "## Using Low Rank Adaptation Technique For Fine-tuning\n",
+        "\n",
+        "Next we setup our LORA finetuning. While giants like Mistral and Phi2 boast impressive capabilities, their computational demands can be hefty. This is where LORA, a lightweight and efficient parameter reduction technique, steps in, offering a compelling alternative.\n",
+        "LORA stands for Low-Rank Optimized Adaptor. It acts as a \"shortcut\" within an LLM, compressing the original model's parameters into a much smaller, low-rank representation. This significantly reduces the model size and computational cost while aiming to maintain similar performance in text generation tasks.\n",
```

Comment:
Adapter not Adaptor

Reply:
There are many wrong statements in this section. Please remove all statements you are not 100% certain of. For instance, LORA does not reduce a model's size, it just optimizes a small set of adapter parameters that are added to the pretrained model

Reply:
QLORA on the other hand does shrink the model by quantization all of its parameters

Reply:
I meant it freezes a layer of parameters, but will update the language.